official/keras_application_models/benchmark_main.py

```diff
@@ -0,0 +1,166 @@
+MODELS = {
+    "vgg16": tf.keras.applications.VGG16,
+    "vgg19": tf.keras.applications.VGG19,
+    "inceptionv3": tf.keras.applications.InceptionV3,
+    "xception": tf.keras.applications.Xception,
+    "resnet50": tf.keras.applications.ResNet50,
+    "inceptionresnetv2": tf.keras.applications.InceptionResNetV2,
+    "mobilenet": tf.keras.applications.MobileNet,
+    "densenet121": tf.keras.applications.DenseNet121,
+    "densenet169": tf.keras.applications.DenseNet169,
+    "densenet201": tf.keras.applications.DenseNet201,
+    "nasnetlarge": tf.keras.applications.NASNetLarge,
+    "nasnetmobile": tf.keras.applications.NASNetMobile,
+}
+
+
+def main(_):
+  # Ensure a valid model name was supplied via command line argument
+  if FLAGS.model not in MODELS.keys():
+    raise AssertionError("The --model command line argument should "
+                         "be a key in the `MODELS` dictionary.")
+
+  # Load the model
+  tf.logging.info("Benchmark on {} model...".format(FLAGS.model))
+  keras_model = MODELS[FLAGS.model]
+  model = keras_model(weights=None)
+
+  # Get dataset
+  dataset_name = "ImageNet"
+  if FLAGS.use_synthetic_data:
+    tf.logging.info("Using synthetic dataset...")
+    dataset_name += "_Synthetic"
+    batch_size = resnet_run_loop.per_device_batch_size(
+        FLAGS.batch_size, flags_core.get_num_gpus(FLAGS))
+    train_num_images = batch_size
+    val_num_images = batch_size
+    train_dataset = dataset.generate_synthetic_input_dataset(
+        FLAGS.model, train_num_images)
+    val_dataset = dataset.generate_synthetic_input_dataset(
+        FLAGS.model, val_num_images)
+  else:
+    # Use the actual ImageNet dataset (TODO)
+    raise ValueError("Only synthetic dataset is supported!")
+
+  # If run with multiple GPUs
+  num_gpus = flags_core.get_num_gpus(FLAGS)
+  if num_gpus > 0:
+    model = tf.keras.utils.multi_gpu_model(model, gpus=num_gpus)
+
+  # Configure the model
+  model.compile(loss="categorical_crossentropy",
+                optimizer="sgd",
+                metrics=["accuracy"])
+
+  # Create benchmark logger for benchmark logging
+  run_params = {
+      "batch_size": FLAGS.batch_size,
+      "synthetic_data": FLAGS.use_synthetic_data,
+      "train_epochs": FLAGS.train_epochs
+  }
+
+  benchmark_logger = logger.config_benchmark_logger()
+  benchmark_logger.log_run_info(
+      model_name=FLAGS.model,
+      dataset_name=dataset_name,
+      run_params=run_params,
+      test_id=FLAGS.benchmark_test_id)
+
+  # Create callbacks that log metric values about the training and evaluation
+  callbacks = model_callbacks.get_model_callbacks(
+      FLAGS.callbacks,
+      batch_size=FLAGS.batch_size,
+      metric_logger=benchmark_logger)
+
+  # Train and evaluate the model
+  history = model.fit(
+      train_dataset,
+      epochs=FLAGS.train_epochs,
+      callbacks=callbacks,
+      validation_data=val_dataset,
+      steps_per_epoch=int(np.ceil(train_num_images / batch_size)),
```

Comment:
why np.ceil is used here? I assume train_num_images and batch_size are just real number, rather than matrix.